api/src/main/java/io/druid/data/input/impl/FileIteratingFirehose.java

```diff
@@ -44,13 +47,26 @@ public FileIteratingFirehose(
   {
     this.lineIterators = lineIterators;
     this.parser = parser;
+    final ParseSpec parseSpec = parser.getParseSpec();
+    if (parseSpec instanceof CSVParseSpec) {
```

Comment:
Is your idea that an indexing mechanism that doesn't support `hasHeaderRow` / `skipHeaderRows` would just ignore them completely? Maybe we could achieve that by having a method like `parser.startFile()` that FileIteratingFirehose calls, but none of the other Parser callers call. And if it's not called, the Parsers would ignore those header row related fields.

Or were you thinking of a different kind of smarter behavior?